river/Cursor.zig

```diff
@@ -444,6 +449,13 @@ const SurfaceAtResult = struct {
 pub fn surfaceAt(self: Self) ?SurfaceAtResult {
     const lx = self.wlr_cursor.x;
     const ly = self.wlr_cursor.y;
+    return surfaceAtPosition(lx, ly);
+}
+
+/// Find the surface on the given position if any, and return information about that
+/// surface and the given position in surface local coords.
+/// This function must be kept in sync with the rendering order in render.zig.
+pub fn surfaceAtPosition(lx: f64, ly: f64) ?SurfaceAtResult {
```

Comment:
Since this is now also used elsewhere in a different context, I am not really a fan of having it in `Cursor.zig`, seems a bit leaky abstraction-wise.

Reply:
I also considered moving it somewhere else, but wasn't sure where.